src/Microsoft.Extensions.Hosting.Abstractions/HostingAbstractionsHostExtensions.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting
+{
+    public static class HostingAbstractionsHostExtensions
+    {
+        /// <summary>
+        /// Starts the host synchronously.
+        /// </summary>
+        /// <param name="host"></param>
+        public static void Start(this IHost host)
+        {
+            host.StartAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Attempts to gracefully stop the host with the given timeout.
+        /// </summary>
+        /// <param name="host"></param>
+        /// <param name="timeout">The timeout for stopping gracefully. Once expired the
+        /// server may terminate any remaining active connections.</param>
+        /// <returns></returns>
+        public static Task StopAsync(this IHost host, TimeSpan timeout)
+        {
+            return host.StopAsync(new CancellationTokenSource(timeout).Token);
+        }
+
+        /// <summary>
+        /// Block the calling thread until shutdown is triggered via Ctrl+C or SIGTERM.
+        /// </summary>
+        /// <param name="host">The running <see cref="IHost"/>.</param>
+        public static void WaitForShutdown(this IHost host)
+        {
+            host.WaitForShutdownAsync().GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Returns a Task that completes when shutdown is triggered via the given token.
+        /// </summary>
+        /// <param name="host">The running <see cref="IHost"/>.</param>
+        /// <param name="token">The token to trigger shutdown.</param>
+        public static async Task WaitForShutdownAsync(this IHost host, CancellationToken token = default(CancellationToken))
```

Comment:
Ad discussed WaitForTokenShutdownAsync can be renamed to WaitForShutdownAsyn in this scenario.